Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py

```diff
@@ -0,0 +1,109 @@
+from CommonServerPython import *
+import sys
+import json
+
+
+def add_entries(single_reply, email_related_incident):
+    """Add the entries to the related incident
+    Args:
+        single_reply: The email reply.
+        email_related_incident: The related incident.
+    Returns:
+
+    """
+    entries_str = json.dumps([{"Type": 1, "ContentsFormat": 'markdown', "Contents": single_reply, "tags": ['email'
+                                                                                                           '-thread']}])
+    res = demisto.executeCommand("addEntries", {"entries": entries_str, 'id': email_related_incident})
+    if is_error(res):
+        demisto.results('ERROR: PreprocessEmail - addEntries: {}'.format(res['Contents']))
+        # Fail safe to not drop emails if we couldn't add entries to existing incident
+        demisto.results(True)
+        sys.exit(0)
+
+
+def set_email_reply(email_from, email_to, email_cc, email_body):
+    """Add the entries to the related incident
+    Args:
+        email_from: The email author mail.
+        email_to: The email recipients.
+        email_cc: The email cc.
+        email_body: The email body.
+    Returns:
+        Email reply.
+
+    """
+    single_reply = f"""
+    From: *{email_from}*
+    To: *{email_to}*
+    CC: *{email_cc}*
+
+
+    ----
+
+    {email_body}
+    """
+    return single_reply
+
+
+def get_incident_by_query(query):
+    """
+    Get incident id and return it's details.
+    Args:
+        query: Query for the incident id.
+    Returns:
+        Incident details.
+    """
+    res = demisto.executeCommand("GetIncidentsByQuery", {"query": query, "Contents": "id,status"})[0]
+
+    if is_error(res):
+        demisto.results('ERROR: PreprocessEmail - GetIncidentsByQuery: {}'.format(res['Contents']))
+        demisto.results(True)
+        sys.exit(0)
+
+    incident_details = json.loads(res['Contents'])[0]
+    return incident_details
+
+
+def check_incident_status(incident_details, email_related_incident):
+    """Get the incident details and checks the incident status.
+    Args:
+        incident_details: The incident details.
+        email_related_incident: The related incident.
+    """
+    status = incident_details.get('status')
+    if status == 2:  # closed
+        res = demisto.executeCommand("reopenInvestigation", {"id": email_related_incident})
+        if is_error(res):
+            demisto.results('ERROR: PreprocessEmail - Reopen incident: {}'.format(res['Contents']))
+            # Fail safe to not drop emails if we couldn't reopen incident
+            demisto.results(True)
+            sys.exit(0)
+
+
+def main():
+    incident = demisto.incidents()[0]
```

Comment:
What happens if we have 0 incidents?

Reply:
When receive an email it creates an incident so there isn't a situation  where there aren't any incidents.